squashfs4: backport fix for glibc version 2.26+

Include sys/sysmacros.h for major/minor/makedev. These functions have
always been defined in sys/sysmacros.h under Linux C libraries.  For
some, including sys/types.h implicitly includes that as well, but glibc
wants to deprecate that, and some others already have.  Include the
header explicitly for the funcs.

Fixes: FS#1017

Signed-off-by: Bob Brooks <gitbugged@cool.fr.nf>
--- a/squashfs-tools/mksquashfs.c	2017-09-24 17:15:26.041555804 -0400
+++ b/squashfs-tools/mksquashfs.c	2017-09-24 17:16:31.124010103 -0400
@@ -49,6 +49,7 @@
 #include <regex.h>
 #include <fnmatch.h>
 #include <sys/wait.h>
+#include <sys/sysmacros.h>
 
 #ifndef linux
 #define __BYTE_ORDER BYTE_ORDER
--- a/squashfs-tools/unsquashfs.c	2017-09-24 17:19:37.137353855 -0400
+++ b/squashfs-tools/unsquashfs.c2	2017-09-24 17:20:33.848303653 -0400
@@ -28,6 +28,7 @@
 #include "read_fs.h"
 #include "compressor.h"
 #include "xattr.h"
+#include <sys/sysmacros.h>
 
 #include <sys/sysinfo.h>
 #include <sys/types.h>
